### DISCUSSION TITLE
Apply QScrollArea in QLayout  Container

### DISCUSSION THREAD

**Discussion by aabm00** [quasarframework/quasar#9568] [General - Components / Directives / etc]
Hello 

I'm trying to apply **QScrollArea**  in a **QLayout Container** in the  **q-page-container** area but I can't get it to work.

I have seen that it is possible to apply it in the "**QDrawer**" as shown in the Basic example. 
[https://next.quasar.dev/layout/drawer#basic](https://next.quasar.dev/layout/drawer#basic)

I would like to do the same but applied to the content part of a QLayout container like this:

```
<q-layout view="lHh Lpr lFf" container>

   ........

   <q-page-container>
     <q-page >

        <q-scroll-area class="fit" >

            <div v-for="n in 105" :key="n">
                 Lorem ipsum dolor sit amet, consectetur adipisicing
                 elit, sed do eiusmod tempor incididunt ut labore et
                 dolore magna aliqua.
           </div>

        </q-scroll-area>

    </q-page>
  </q-page-container>

  .........

</q-layout>
```

It doesn't work I would appreciate any advice

Thanks

**Comment by ontwikkelfabriek** [comment]
Your scrollarea does not have a height try:

```
 <q-scroll-area style="height: 100vh" >
```

**Reply by aabm00** [reply]
Hi @ontwikkelfabriek

Thanks for your answer.

In fact,, It is a **Qlayout container** (a QLayout container inside the main Qlayout), so its dimensions are not those of the Viewport (relative to the size of the root container of the page), so it is not the height of the **Qlayout container**. 

Applying `<q-scroll-area style="height: 100vh" >` displays 2 scroll bars, one from the **QScrollArea** (does not scroll the full height of the content) and one from the native scroll (scrolls the height that the QScrollArea does not scroll).

In fact with <`q-scroll-area class="fit" >` is applying a height because the **quasar** class "**fit**" is **width = 100% and height = 100%** relative to the parent container which in this case would be the **Qlayout container**. It should work the same as it works in the **Qdrawer** in the **example of the link**.

[https://next.quasar.dev/layout/drawer#basic](https://next.quasar.dev/layout/drawer#basic)

Obviously there is some detail that escapes me.

**Comment by aabm00** [comment]
Hi !

A possible solution based on @ontwikkelfabriek's answer could be the one shown below, but it's not practical since we have to make calculations based on whether or not it incorporates **header**, **footer** or **padding**.

So I would like to know if there is a way to make it work by applying the quasar **fit** class (as shown in the main question)
or manually working with **width :100%** and **height: 100%.** in the same way it works in the drawer in the example
[https://next.quasar.dev/layout/drawer#basic](https://next.quasar.dev/layout/drawer#basic)

`<q-scroll-area class= my-scroll-area" >`

```
<style lang="scss">
  .my-scroll-area {
    height: calc(100vh - ( header + footer + padding * 2 ) 
  }
</style>
```

Thanks
